Chocobozzz/PeerTube
# add user account email verificiation

Two remaining items related to this to (potentially) add in future PR:

1. Ability for admin to verify users manually via admin console
2. If SMTP is not configured properly the user will get a message indicating the email successfully sent. This is the same with reset-password. Only the server admin will see via logs. Not sure if this intended, but I could add some middleware to report on SMTP-related errors in the front-end. Let me know

Includes server and client code to:

* enable verificationRequired via custom config
* send verification email with registration
* ask for verification email
* verify via email
* prevent login if not verified and required
* conditional client links to ask for new verification email
